app/src/main/java/com/claraberriel/mvpparking/mvp/presenter/ReservationPresenter.java

```diff
@@ -0,0 +1,108 @@
+package com.claraberriel.mvpparking.mvp.presenter;
+
+import android.util.Log;
+
+import com.claraberriel.mvpparking.entities.Parking;
+import com.claraberriel.mvpparking.entities.Reservation;
+import com.claraberriel.mvpparking.mvp.model.ReservationModel;
+import com.claraberriel.mvpparking.mvp.view.ReservationView;
+
+import java.util.Date;
+
+public class ReservationPresenter {
+    ReservationView reservationView;
+    ReservationModel reservationModel;
+
+    public ReservationPresenter(ReservationView reservationView, ReservationModel reservationModel) {
+        this.reservationView = reservationView;
+        this.reservationModel = reservationModel;
+    }
+
+    public Parking getParkingWithReservations() {
+        return reservationModel.getParking();
+    }
+
+    /**
+     * Event Handlers
+     */
+
+    public void onFromClicked() {
+        reservationView.startDateTimeDialog();
+    }
+
+    public void onToClicked() {
+        reservationView.endDateTimeDialog();
+    }
+
+    public boolean onSchedule() {
+        if (areDatesValid() && isParkingNumberValid() && isSecurityCodeValid()) {
+            Reservation reservation = new Reservation(reservationView.getStartDate().getTime(),
```

Comment:
I would suggest removing all this logic and do something like

`reservationModel.addReservation(reservationView.getStartDate().getTime(), reservationView. getEndDate().getTime())`

the code will look cleaner as the model already has part of the information

Reply:
I get a bunch of errors if I try that, my addReservation method takes a reservation as a parameter, but I need to call the Reservation constructor to give it all the data (dates, parking number, code). Best I could do is: reservationModel.addReservation(new Reservation(startDate, endDate, blah, blah..) so I have one less line of code by removing the variable declaration.